Sources/WeatherSDK/EndpointResolver.swift

```diff
@@ -0,0 +1,132 @@
+// Code generated by smithy-swift-codegen. DO NOT EDIT!
+
+import ClientRuntime
+
+public struct EndpointParams {
+    /// docs
+    public let stage: Swift.String?
+
+    public init(
+        stage: Swift.String? = nil
+    )
+    {
+        self.stage = stage
+    }
+}
+
+public protocol EndpointResolver {
+    func resolve(params: EndpointParams) throws -> ClientRuntime.Endpoint
+}
+
+public struct DefaultEndpointResolver: EndpointResolver  {
+
+    private let engine: ClientRuntime.EndpointsRuleEngine
+    private let ruleSet = "{\"version\":\"1.3\",\"parameters\":{\"Stage\":{\"required\":false,\"documentation\":\"docs\",\"type\":\"String\"}},\"rules\":[{\"conditions\":[{\"fn\":\"isSet\",\"argv\":[{\"ref\":\"Stage\"}]},{\"fn\":\"stringEquals\",\"argv\":[{\"ref\":\"Stage\"},\"alpha\"]}],\"endpoint\":{\"url\":\"https://fpmhmw9kd7.execute-api.us-west-2.amazonaws.com/beta\",\"properties\":{},\"headers\":{}},\"type\":\"endpoint\"},{\"conditions\":[{\"fn\":\"isSet\",\"argv\":[{\"ref\":\"Stage\"}]},{\"fn\":\"stringEquals\",\"argv\":[{\"ref\":\"Stage\"},\"beta\"]}],\"endpoint\":{\"url\":\"https://fpmhmw9kd7.execute-api.us-west-2.amazonaws.com/beta\",\"properties\":{},\"headers\":{}},\"type\":\"endpoint\"},{\"conditions\":[{\"fn\":\"isSet\",\"argv\":[{\"ref\":\"Stage\"}]},{\"fn\":\"stringEquals\",\"argv\":[{\"ref\":\"Stage\"},\"gamma\"]}],\"endpoint\":{\"url\":\"https://mow32it2pa.execute-api.us-east-1.amazonaws.com/gamma\",\"properties\":{},\"headers\":{}},\"type\":\"endpoint\"},{\"conditions\":[{\"fn\":\"isSet\",\"argv\":[{\"ref\":\"Stage\"}]},{\"fn\":\"stringEquals\",\"argv\":[{\"ref\":\"Stage\"},\"prod\"]}],\"endpoint\":{\"url\":\"https://bqcgz0nn7a.execute-api.us-east-1.amazonaws.com/prod\",\"properties\":{},\"headers\":{}},\"type\":\"endpoint\"},{\"conditions\":[],\"endpoint\":{\"url\":\"https://bqcgz0nn7a.execute-api.us-east-1.amazonaws.com/prod\",\"properties\":{},\"headers\":{}},\"type\":\"endpoint\"}]}"
```

Comment:
What exactly are the hosts pointed to in these rules?

Reply:
I was just verifying the endpoint rulesSet gets generated correctly using fake urls. I will remove this.